packages/teams-js/prepBetaRelease.js

```diff
@@ -128,37 +128,46 @@ function getNewNextDevSuffixNum(currNextDevVer, currPkgJsonVer) {
     // If the next-dev beta suffix is not equal to or exactly 1 higher than the package.json beta suffix, throw an error since this does not follow our versioning structure.
     // e.g. package.json version: 2.0.0-beta.1 and current next-dev version: 2.0.0-beta.0-dev.0 should throw an error.
   } else if (betaSuffixNumInCurrNextDev !== betaSuffixNumInPkgJsonVer) {
-    throw new Error(`Invalid beta version suffix number ${betaSuffixNumInPkgJsonVer} in package.json version ${currPkgJsonVer}`);
+    throw new Error(
+      `Invalid beta version suffix number ${betaSuffixNumInPkgJsonVer} in package.json version ${currPkgJsonVer}`,
+    );
   }
   return newNextDevVerSuffixNum;
 }
 
 /**
- * Generates the new package.json content with updated next-dev version number. The version number is
+ * Generates the new package.json content with updated beta version number. The version number is
  * the only thing that's changed.
- * @param currNextDevVer The current next-dev version to bump the new next-dev version number from. (e.g. 2.0.0-beta.1-dev.0)
+ * @param currBetaVer The current beta version to bump the new beta version number from. (e.g. 2.0.0-beta.1)
  * @returns the new package.json content in JSON format.
  */
-function getNewPkgJsonContent(currNextDevVer) {
+function getNewPkgJsonContent(currBetaVer) {
   const packageJson = getPackageJson();
 
   // get package version from package.json
   let currPkgJsonVer = getPkgJsonVersion(packageJson);
+  let currBetaPrefix = getPrefix(currBetaVer);
 
   console.log('package.json version: ' + currPkgJsonVer);
-  console.log('current next-dev tagged version: ' + currNextDevVer);
+  console.log('current beta tagged version: ' + currBetaVer);
 
-  if (currPkgJsonVer.includes('dev')) {
-    throw new Error(`The given package.json\'s version ${currPkgJsonVer} contains the substring \'dev\' which is reserved for non-prod versions. Please fix the package.json version first in order to allow for proper version incrementation.`);
+  if (currPkgJsonVer.includes('beta')) {
+    throw new Error(
+      `The given package.json\'s version ${currPkgJsonVer} contains the substring \'beta\' which is reserved for non-prod versions. Please fix the package.json version first in order to allow for proper version incrementation.`,
+    );
+  }
+  let betaVerNum = 0;
+  if (currBetaPrefix === currPkgJsonVer) {
+    betaVerNum = getSpecificVerSuffixNum('beta', currPkgJsonVer) + 1;
```

Comment:
If you pass in "2.1.0-beta.7" as the input to `currBetaVer`, then this script will return `2.1.0-beta.0`, which is incorrect. It should be `2.1.0-beta.8`, based on the input. 

The script needs to take into account whether to increment the version that would come from `yarn beachball bump` OR increment the `currBetaVersion`. I think you need to update `getNewNextDevSuffixNumber` to take the versions into account to determine how to update the beta suffix number like the script originally had on old line 160. (This is related to my comment on old line 131.)